node-persistent-cache.cpp

```diff
@@ -532,10 +531,10 @@ int node_persistent_cache::get_list(nodelist_t &out, const idlist_t nds)
 {
     out.assign(nds.size(), osmNode());
 
-    bool need_fetch = false;;
+    bool need_fetch = false;
     for (size_t i = 0; i < nds.size(); ++i) {
         /* Check cache first */
-        if (ram_cache && (ram_cache->get(&out[i], nds[i]) != 0)) {
+        if (!ram_cache || (ram_cache->get(&out[i], nds[i]) == 0)) {
```

Comment:
This line is the one causing the problems described in https://github.com/openstreetmap/osm2pgsql/issues/351#issuecomment-105362202